pkg/sql/logictest/testdata/logic_test/fk

```diff
@@ -3671,3 +3671,51 @@ ALTER TABLE child_59582 ADD FOREIGN KEY (i) REFERENCES child_59582(rowid)
 
 statement ok
 DROP TABLE parent_59582, child_59582
+
+subtest reference_constraint_different_order
+
+# https://github.com/cockroachdb/cockroach/issues/63324
+# Referencing an existing constraint by fields in a different order.
+# Should still find existing constraint.
+
+statement ok
+CREATE TABLE reference_constraint_different_order_parent (
+  a INT8 NOT NULL,
+  b INT8 NOT NULL,
+  c INT8 NOT NULL,
+  d VARCHAR(10),
+  PRIMARY KEY (a, b, c)
+);
+CREATE TABLE reference_constraint_different_order_child (
+  id INT8,
+  a INT8,
+  b INT8,
+  c INT8,
+  e VARCHAR(10)
+);
+
+statement ok
+ALTER TABLE reference_constraint_different_order_child
+  ADD CONSTRAINT p_fk FOREIGN KEY (a, b, c) REFERENCES reference_constraint_different_order_parent (a, c, b);
+
+statement ok
+INSERT
+INTO
+	reference_constraint_different_order_parent
+VALUES
+	(1, 2, 3, 'par_val');
+INSERT
+INTO
+	reference_constraint_different_order_child
+VALUES
+	(99, 1, 2, 3, 'child_val');
```

Comment:
This insert should not be successful, it should be a FK violation. There is no row in the parent where `(a, c, b) = (1, 2, 3)`. I think the issue is that we are setting `ReferencedColumnIDs` to the result of `targetDesc.GetExplicitColumnIDs()` without ordering these columns according to the column ordering of the `ALTER TABLE .. ADD CONSTRAINT .. FOREIGN KEY` statement.

Feel free to try to solve this, if you'd like. We should have tests for both a successful INSERT and a failed INSERT due to a FK violation. I'm also happy to take over the PR from here.

Reply:
I now realise that my assumption that this should work was why I went off on the tangent of using the existing PK definition with `GetExplicitColumnIDs()` in the first place. But you're right, it shouldn't work, because the reference is `b -> c` and vice versa, not `b -> b` etc.

Refreshing that assumption, I believe I was actually done a while back, before I went on this tangent.
I will fix the test expectations, then remove the unnecessary changes.

Reply:
If indeed that is the intention?

Reply:
@mgartner subject to that, I've added a test which checks the correct behaviour, and one which proves a violation otherwise. Let me know if those tests match what you were expecting 😄